### PR TITLE
ivs font writing support

### PR DESCRIPTION
**Pull request by doobo**
[issues#387](https://github.com/LibrePDF/OpenPDF/issues/387),IVS font support, irregular ideographic writing support.

### CODE REVIEW

**Comment by mkl-public** [comment]
Looks much better than the previous attempts.
I cannot assess the actual IVS support, not my field of expertise.

**Comment by andreasrosdal** [comment]
Thanks for the contribution. Can you please add a unit test for this?

File: openpdf/src/main/java/com/lowagie/text/pdf/TrueTypeFontUnicode.java [hunk 1]
```diff
@@ -541,4 +547,35 @@ public boolean setCharAdvance(int c, int advance) {
...
     }
+
+	/**
+	 *
+	 * ivs character to hexadecimal
+	 * @param char1
+	 * @param char2
+	 * @return
+	 */
+    private String toHex(int char1, int char2) {
```
Comment (andreasrosdal) [inline openpdf/src/main/java/com/lowagie/text/pdf/TrueTypeFontUnicode.java:558]: Is there a built-in function in Java which does this?

File: openpdf/src/main/java/com/lowagie/text/pdf/FontDetails.java [hunk 2]
```diff
@@ -214,10 +214,50 @@ BaseFont getBaseFont() {
                         String s = new String(glyph, 0, i);
                         b = s.getBytes(CJKFont.CJK_ENCODING);
 
-                    }
-                    else {
+                    } else {
+                        //ivs font handler,Simply judge whether it is IVS font or not
+                        if(text != null && text.contains("\udb40")){
```
Comment (codecracker2014) [inline openpdf/src/main/java/com/lowagie/text/pdf/FontDetails.java:219]: We can decide based on only \udb40?

File: openpdf/src/main/java/com/lowagie/text/pdf/FontDetails.java [hunk 3]
```diff
@@ -214,10 +214,50 @@ BaseFont getBaseFont() {
                         String s = new String(glyph, 0, i);
                         b = s.getBytes(CJKFont.CJK_ENCODING);
 
-                    }
-                    else {
+                    } else {
+                        //ivs font handler,Simply judge whether it is IVS font or not
```
Comment (codecracker2014) [inline openpdf/src/main/java/com/lowagie/text/pdf/FontDetails.java:218]: It would be better if we can keep ivs handler outside of this class.

**Pull request closed without merging**